File: V1/search insert position.py
class Solution(object):
    def searchInsert(self, nums, target):
        low, high = 0, len(nums) - 1

        while low <= high:
            mid = low + ((high - low) // 2)

            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                low = mid + 1
            else:
                high = mid - 1

        return low
        
Solucion = Solution()

NUMBERS = [1,2,3,5,6]
target = 4

print(Solucion.searchInsert(NUMBERS,target))

File: V1/Search Insert Position.cpp
#include <iostream>
using namespace std;

#include <vector>

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        int low = 0;
        int high = nums.size()-1;

        while (low <=high){
            int mid = low + ((high-low)/2);

            if(nums[mid] == target){
                return mid;

            } else if(nums[mid] < target){
                low = mid+1;

            } else{
                high = mid-1;
            }
        }
        return low;
    }

};


int main(){

    Solution Hola;

    vector<int> Lista= {1,3,5,6,7,8,9};

    cout << Hola.searchInsert(Lista,7);

    return 0;
}